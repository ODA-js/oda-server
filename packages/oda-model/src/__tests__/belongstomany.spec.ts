import 'jest';
import { BelongsToMany } from '../';

describe('belongstomany ', () => {
  it('loads', () => {
    const bt = new BelongsToMany({
      belongsToMany: 'name#id',
      entity: 'user',
      field: 'name',
      using: 'C',
    });
    expect(bt).toMatchSnapshot('raw');
    expect(bt.toObject()).toMatchSnapshot('toObject');
  });
});
