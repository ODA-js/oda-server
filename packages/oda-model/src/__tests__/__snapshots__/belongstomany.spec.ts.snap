// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`belongstomany  loads: raw 1`] = `
BelongsToMany {
  "$obj": Object {
    "belongsToMany": EntityReference {
      "$obj": Object {
        "backField": "id",
        "entity": "Name",
        "field": "id",
        Symbol(metadata): Object {},
      },
    },
    "entity": "user",
    "field": "name",
    "fields": Map {},
    "using": EntityReference {
      "$obj": Object {
        "backField": "id",
        "entity": "C",
        "field": "id",
        Symbol(metadata): Object {},
      },
    },
    Symbol(metadata): Object {
      "name": Object {
        "full": "userBelongsToManyNames",
        "normal": "userNames",
        "short": "Names",
      },
      "persistence": Object {
        "embedded": false,
        "single": false,
        "stored": false,
      },
      "verb": "BelongsToMany",
    },
  },
}
`;

exports[`belongstomany  loads: toObject 1`] = `
Object {
  "belongsToMany": "id@Name#id",
  "embedded": false,
  "entity": "user",
  "field": "name",
  "fields": Array [],
  "metadata": Object {
    "name": Object {
      "full": "userBelongsToManyNames",
      "normal": "userNames",
      "short": "Names",
    },
    "persistence": Object {
      "embedded": false,
      "single": false,
      "stored": false,
    },
    "verb": "BelongsToMany",
  },
  "name": undefined,
  "opposite": undefined,
}
`;
